apps/finance/app/src/components/useDownloadData.js

```diff
@@ -0,0 +1,95 @@
+import { useContext, useCallback } from 'react'
+import { format } from 'date-fns'
+import { saveAs } from 'file-saver'
+import { useCurrentApp } from '@aragon/api-react'
+import { useToast } from '@aragon/ui'
+import { IdentityContext } from './IdentityManager/IdentityManager'
+import { formatDate, YYMMDD_LONG_FORMAT } from '../lib/date-utils'
+import { formatTokenAmount } from '../lib/utils'
+import { toChecksumAddress } from '../lib/web3-utils'
+
+// Transforms a two dimensional array into a CSV data structure
+// Surround every field with double quotes + escape double quotes inside.
+function csv(data) {
+  return data
+    .map(cells => cells.map(cell => `"${cell.replace(/"/g, '""')}"`).join(','))
+    .join('\n')
+}
+
+const getDownloadFilename = (appAddress, { start, end }) => {
+  const today = format(Date.now(), YYMMDD_LONG_FORMAT)
+  let filename = `finance_${appAddress}_${today}.csv`
+  if (start && end) {
+    const formattedStart = format(start, YYMMDD_LONG_FORMAT)
+    const formattedEnd = format(end, YYMMDD_LONG_FORMAT)
+    filename = `finance_${appAddress}_${formattedStart}_to_${formattedEnd}.csv`
+  }
+  return filename
+}
+
+const getDownloadData = async (transfers, tokenDetails, resolveAddress) => {
+  const processedTransfers = await Promise.all(
+    transfers.map(
+      async ({
+        date,
+        numData: { amount },
+        reference,
+        isIncoming,
+        entity,
+        token,
+      }) => {
+        const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+        const formattedAmount = formatTokenAmount(
+          amount,
+          isIncoming,
+          decimals,
+          true,
+          { rounding: 5 }
+        )
+        const { name = '' } = (await resolveAddress(entity)) || {}
+        return [
+          formatDate(date),
```

Comment:
This is currently using `MM/dd/yy`, but perhaps we should default the date formatting to `dd/mm/yy` instead? That's what it used to be: https://github.com/aragon/aragon-apps/pull/1103/files#diff-c6b357d815fc23c7aea312476aabe3beL33.

Or building on the offline discussion, maybe we should just settle on yyyy-mm-dd as the default. It's kind of weird we use that format for the file name but then not the dates inside.

Reply:
Yes, we agreed yesterday to implement the ISO international format `yyyy-mm-dd` by default everywhere.

Reply:
As a result of this, we need to update the tables and DataView; This will be done in a separate PR.